jmetal-core/src/main/java/org/uma/jmetal/util/experiment/component/EvolutionChart/HistoryData.java

```diff
@@ -0,0 +1,162 @@
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package org.uma.jmetal.util.experiment.component.EvolutionChart;
+
+import org.uma.jmetal.util.experiment.component.GenerateEvolutionChart;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.uma.jmetal.qualityindicator.impl.Epsilon;
+import org.uma.jmetal.qualityindicator.impl.InvertedGenerationalDistancePlus;
+import org.uma.jmetal.qualityindicator.impl.Spread;
+import org.uma.jmetal.qualityindicator.impl.hypervolume.PISAHypervolume;
+import org.uma.jmetal.solution.DoubleSolution;
+import org.uma.jmetal.util.experiment.Experiment;
+import org.uma.jmetal.util.experiment.util.ExperimentAlgorithm;
+import org.uma.jmetal.util.front.Front;
+import org.uma.jmetal.util.front.imp.ArrayFront;
+import org.uma.jmetal.util.front.util.FrontNormalizer;
+import org.uma.jmetal.util.front.util.FrontUtils;
+import org.uma.jmetal.util.point.util.PointSolution;
+
+/**
+ *
+ * @author Lucas Prestes <lucas.prestes.lp@gmail.com> 
+ */
+public class HistoryData {
+    public static final String DEFAULT_BASE = "history";
+    private Integer testId;
+    
+    private Integer numberOfData;
+    private Integer numberOfTest;
+    
+    /**
+     * Number of data (%) / number of test
+     */
+    private Double[][] history;
+   
+
+    public HistoryData(Integer numberOfTest) {
+        this.testId = -1;
+        this.numberOfData = 100;
+        this.numberOfTest = numberOfTest;
+        history = new Double[this.numberOfData][this.numberOfTest];
+
+        for (int i = 0; i < this.numberOfData; i++) {
+            for (int j = 0; j < this.numberOfTest; j++) {
+                history[i][j] = 0.0;
```

Comment:
If you never have null values, you may instead use  the primitive type `double` instead of its object counterpart. It would also allow you to avoid the initialisation phase (a `double` is by default at `0.0`).